SmartSchedule.Application/Event/Commands/UpdateEvent/UpdateEventCommand.cs

```diff
@@ -0,0 +1,68 @@
+﻿namespace SmartSchedule.Application.Event.Commands.UpdateEvent
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using MediatR;
+    using SmartSchedule.Application.Exceptions;
+    using SmartSchedule.Persistence;
+    using ValidationException = FluentValidation.ValidationException;
+
+    public class UpdateEventCommand : IRequest
+    {
+        public int Id { get; set; }
+        public DateTime StartDate { get; set; }
+        public DateTime EndTime { get; set; }
+        public DateTime ReminderAt { get; set; }
+        public string Name { get; set; }
+        public int RepeatsEvery { get; set; }
+        public string Longitude { get; set; }
+        public string Latitude { get; set; }
+
+        public class Handler : IRequestHandler<UpdateEventCommand, Unit>
+        {
+            private readonly SmartScheduleDbContext _context;
+
+            public Handler(SmartScheduleDbContext context)
+            {
+                _context = context;
+            }
+            public async Task<Unit> Handle(UpdateEventCommand request, CancellationToken cancellationToken)
+            {
+                var entityEvent = await _context.Events.FindAsync(request.Id);
+
+                if (entityEvent == null)
+                {
+                    throw new NotFoundException("Event", request.Id);
+                }
+
+                var vResult = await new UpdateEventCommandValidator(_context).ValidateAsync(request, cancellationToken);
+
+                if (!vResult.IsValid)
+                {
+                    throw new ValidationException(vResult.Errors);
+                }
+
+                var entityLocation = new Domain.Entities.Location
+                {
+                    Latitude = request.Latitude,
+                    Longitude = request.Longitude
+                };
+                _context.Locations.Add(entityLocation);
```

Comment:
Napewno to dziala ? przetestowalas na wiekszej ilosci danych ?

Reply:
no kurde testy dzialaja jak chcesz to moge spowalniac kod

Reply:
juz